swarm/storage/resource.go

```diff
@@ -237,7 +237,9 @@ func (self *ResourceHandler) IsValidated() bool {
 func (self *ResourceHandler) keyDataHash(key Key, data []byte) common.Hash {
 	hasher := self.hashPool.Get().(SwarmHash)
 	defer self.hashPool.Put(hasher)
-	hasher.Reset()
+	length := make([]byte, 8)
+	binary.LittleEndian.PutUint64(length, uint64(len(key)+len(data)))
+	hasher.ResetWithLength(length)
```

Comment:
I don't think it's correct to hash these with length. Resource update chunks are different from normal ones.

Reply:
Ok, since I left SHA3 for resource hashing anyway, I reverted this change.
However, I changed this to ResetWithLength according to @zelig 's instructions, so please guys discuss this, I don't know if we need it or not.